.github/workflows/update-dependancies.yaml

```diff
@@ -0,0 +1,43 @@
+name: Update dependencies
+on:
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    - cron:  '30 1 * * *'
+
+jobs:
+  update:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: ./.github/workflows/setup-bazel
+
+      ########################################
+      # Once per night, update dependencies and completely delete and recreate bazel cache
+      ########################################
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - uses: snok/install-poetry@v1
+      - name: If running on schedule (at night) and on main, completely delete bazel cache then update deps
```

Comment:
We also need to run bazel build in this schedule, otherwise the bazel cache will not contain bazel dir and the first build in the day will take very long time.

Reply:
Yup, my bad, fixed